wordpress-mobile/WordPress-Android
# Issue/show onboarding dialog on app start

This PR shows BP onboarding dialog on application start. 


- Dialog displayed only once.
- If you access onboarding dialog before it was supposed to be displayed don app start, it will be marked as displayed, and will not appear on app start.
- You can manually override the flag to test it if necessary.
- Dialog has the lowest priority when it comes to things shown at startup.

I also fixed couple of things in onboarding dialog:
- We get the current site after pressing on a button. Otherwise site might not be initialized when dialog VM starts.
- If we don't have a prompt when you press the Try Now button, we show a snack that tells you that prompts are being loaded. We don't load prompts at this point because they are already being loaded from dashboard.

In addition:
- Minor change to the snackbar sequencer that allows it to work with different type of context.

To test 1:
- To replicate the FF being on by default, set its default value to true in `BloggingPromptsFeatureConfig`.
- Perform a fresh application installation and login into your wpcom account.
- You will probably see the Quick Start prompt on first login instead of BP onboarding.
- Restart the app and confirm that BP onboarding is visible, and behaves as expected.
- Restart the app again, and confirm that the BP onboarding is not visible.

To test 2:
- To replicate the FF being on by default, set its default value to true in `BloggingPromptsFeatureConfig`.
- Perform a fresh application installation and login into your wpcom account.
- You will probably see the Quick Start prompt on first login instead of BP onboarding.
- Instead of restarting the app, open the onboarding dialog from any of the entry points available (card, compact card, etc)
- Restart the app again, and confirm that the BP onboarding is not visible.

To test 3:
- Make sure that BP ff flag is on.
- Install this branch on top of trunk and confirm that BP onboarding dialog is displayed and works as expected.
- Restart the app again, and confirm that the BP onboarding is not visible.

To test 4:
- This one is hard to replicate, but you can trigger the condition manually - [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16723/files#diff-de35a6f81f5ba6284232a93e5ea2eb8ce5b20efd8ab9f3abcda94d3cb69db231R63) set the blogging prompt to null.
- Trigger onboarding dialog in any of the ways described above.
- Press on Try It Now button.
- Confirm that the snackbar is visible.

To test 5 (changes to snackbar sequencer):
- Navigate to comments list, go into comment details and make it as spam.
- Confirm that the snackbar is displayed.

## Regression Notes
1. Potential unintended areas of impact
- BP onboarding is visible at the wrong timing.
- Potential issue with context when showing snackbars.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing. Main activity and context related stuff is very tricky to test automatically.

3. What automated tests I added (or what prevented me from doing so)
- Testing marking BP dialog as displayed when opening it.

PR submission checklist:
